test/Microsoft.TestPlatform.Common.UnitTests/ExtensionFramework/Utilities/TestExtensionsTests.cs

```diff
@@ -239,4 +241,110 @@ public void GetExtensionsDiscoveredFromAssemblyShouldReturnTestDiscoveresAndLogg
         };
         CollectionAssert.AreEqual(expectedLoggers.Keys, extensions.TestLoggers.Keys);
     }
+
+    [TestMethod]
+    public void MergedDictionaryOfEmptyDictionariesShouldBeAnEmptyDictionary()
+    {
+        var first = new Dictionary<string, HashSet<string>>();
+        var second = new Dictionary<string, HashSet<string>>();
+        var merged = TestExtensions.CreateMergedDictionary(first, second);
+
+        // Merging two empty dictionaries should result in an empty dictionary.
+        Assert.IsTrue(merged.Count == 0);
+
+        // Make sure the method is "pure" and returns a new reference.
+        Assert.IsFalse(ReferenceEquals(merged, first));
+        Assert.IsFalse(ReferenceEquals(merged, second));
+    }
+
+    [TestMethod]
+    public void MergedDictionaryOfOneEmptyAndOneNotEmptyDictionaryShouldBeNotEmpty()
```

Comment:
should be not empty is not really what you want it to do.  ShouldContainAllTheItemsOfTheNonEmptyDictionary is more like it. 

also use Fluent assertions and should be equivalent to simplify the check.